ACE/ace/Process_Manager.cpp

```diff
@@ -157,18 +151,10 @@ ACE_Process_Manager::instance (ACE_Process_Manager *tm)
   // Register with the Object_Manager so that the wrapper to
   // delete the proactor will be called when Object_Manager is
   // being terminated.
-
-#if defined ACE_HAS_SIG_C_FUNC
-  ACE_Object_Manager::at_exit (ACE_Process_Manager::instance_,
-                                ACE_Process_Manager_cleanup,
-                                0,
-                                typeid (*ACE_Process_Manager::instance_).name ());
-#else
   ACE_Object_Manager::at_exit (ACE_Process_Manager::instance_,
-                                ACE_Process_Manager::cleanup,
+                                ACE_PROCESS_MANAGER_CLEANUP_FUNCTION,
                                 0,
-                                typeid (*ACE_Process_Manager::instance_).name ());
-#endif /* ACE_HAS_SIG_C_FUNC */
+                                typeid (*t).name ());
```

Comment:
```suggestion
                                typeid (*tm).name ());
```

Reply:
I first had that also, but when this method is called ACE_Process_Manager::instance_ still points to the value it had, where tm points to the new value. I doubt if the code is correct, but using t matches the current behavior

Reply:
So `at_exit` has the wrong pointer?  I agree that fixing it would be outside. the scope of this PR.

Reply:
Yes, that looks the case, but something that is outside of the scope, will create a new issue so that this is on the todo list